### PR TITLE
fs: nvs: introduce nvs_mount and deprecate nvs_init

### LINKED ISSUE #43229: nvs: change nvs_init to accept a device reference
## Introduction

`nvs_init()` now accepts the name of a flash device, forcing the usage of `device_get_binding()`. This adds unnecessary runtime overhead when the flash device can be obtained at compile time, e.g. with `DEVICE_DT_GET()`.

### Problem description

The NVS API is marked as stable, the proposed change would break all direct users of the NVS API. However, the change has low impact (a single API call).

### Proposed change

Change `nvs_init()` to be like:

```c
/**
 * @brief nvs_init
 *
 * Initializes a NVS file system in flash.
 *
 * @param fs Pointer to file system
 * @param dev Pointer to flash device
 * @retval 0 Success
 * @retval -ERRNO errno code if error
 */
int nvs_init(struct nvs_fs *fs, const struct device *dev);
```

## Detailed RFC

https://github.com/zephyrproject-rtos/zephyr/pull/43220
https://github.com/zephyrproject-rtos/zephyr/pull/43260

### Proposed change (Detailed)

N/A

### Dependencies

Settings NVS backend

### Concerns and Unresolved Questions

N/A

## Alternatives

- Deprecate `nvs_init()`
- Create `nvs_mount()`, an equivalent to `nvs_init()` that expects `.flash_device` to be initialized.

### PR DESCRIPTION
**Pull request by gmarull**
Add a new API call to replace nvs_init: nvs_mount. The new API does the
same as nvs_init except that it assumes to be provided with a valid
flash device via `struct nvs_fs` `flash_device` field. Previously, it
was not possible to avoid the runtime overhead of device_get_binding()
even if the flash device was known at compile time.

Resolves https://github.com/zephyrproject-rtos/zephyr/issues/43229
Alternative to https://github.com/zephyrproject-rtos/zephyr/pull/43220

### CODE REVIEW

File: subsys/fs/nvs/nvs.c [hunk 1]
```diff
@@ -856,12 +856,6 @@ int nvs_init(struct nvs_fs *fs, const char *dev_name)
```
Comment (Laczen) [inline subsys/fs/nvs/nvs.c:856]: Maybe a check for `device_is_ready()` should be added here.

**Review by Laczen — DISMISSED** [review]
Probably some extra checks could be added, but this can be done later also.

**Review by de-nordic — DISMISSED** [review]
Looks OK.

@gmarull Can you add note to the issue or log issue for adding release note for the change? There is no template yet for the next release so, unless you are willing to start one, we should take a note that this is required.

**Comment by gmarull** [comment]
@Laczen @de-nordic added release notes

File: doc/releases/release-notes-3.1.rst [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+Documentation
+*************
+
+Tests and Samples
+*****************
+
+Issue Related Items
+*******************
+
+These GitHub issues were addressed since the previous 2.7.0 tagged
```
Comment (de-nordic) [inline doc/releases/release-notes-3.1.rst:172]: 3.0.0 was last tagged, wasn't it?

Comment (gmarull) [reply doc/releases/release-notes-3.1.rst]: oops copy paste artifact, fixed

**Review by de-nordic — APPROVED** [review]
Looks OK.

**Review by Laczen — APPROVED** [review]
Didn't I approve this already ?

**Review by ycsin — APPROVED** [review]
hawkBit part looks good

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
> Didn't I approve this already ?

Yeah, reviews get dismissed on force-push now :(

**Review by nvlsianpu — APPROVED** [review]

**Comment by nvlsianpu** [comment]
@carlescufi This can be merged now.

**Pull request merged**